From 9a7083e4cd8368dbc4e6be480544459513d6be16 Mon Sep 17 00:00:00 2001
From: Matt Johnston <matt@ucc.asn.au>
Date: Tue, 16 Jun 2026 10:21:05 +0800
Subject: [PATCH 10/23] Disallow -t (two factor) with -B (blank passwords)

The blank password option took precedence, the options don't really make
sense together.

(cherry picked from commit 23ec78285639edf068d86bb96f91cba755039740)
---
 src/svr-runopts.c | 3 +++
 1 file changed, 3 insertions(+)

--- a/src/svr-runopts.c
+++ b/src/svr-runopts.c
@@ -468,6 +468,9 @@ void svr_getopts(int argc, char ** argv)
 	if (svr_opts.multiauthmethod && svr_opts.noauthpass) {
 		dropbear_exit("-t and -s are incompatible");
 	}
+	if (svr_opts.multiauthmethod && svr_opts.allowblankpass) {
+		dropbear_exit("-t and -B are incompatible");
+	}
 
 	if (strlen(svr_opts.authorized_keys_dir) == 0) {
 		dropbear_exit("Bad -D");
